disorder-core/src/Disorder/Core/NaturalInt.hs

```diff
@@ -0,0 +1,15 @@
+module Disorder.Core.NaturalInt (
+    NaturalInt (..)
+  ) where
+
+import           Control.Applicative
+
+import           Test.QuickCheck
+
+
+data NaturalInt = NaturalInt {
+    value :: Int
+  } deriving (Eq, Show)
+
+instance Arbitrary NaturalInt where
+  arbitrary = NaturalInt <$> (suchThat arbitrary ((<=) 0))
```

Comment:
Actually an `Integral` instance is kind of lying with these types, but i think `choose` is still better with the upper bound being given by the `Int` `Bounded` instance